### PR TITLE
[cleanup][windows] remove obsolete msys and mingw binary addon dependencies

### PR DESCRIPTION
**Pull request by Rechi**
## Description
- msys dummy dependency should have never been added, the binary addons have to be fixed instead
- mingw got removed, because FFmpeg and libdvd are now build with msvc

## Checklist:
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by Paxxi — APPROVED** [review]
Looks good to me

**Comment by Rechi** [comment]
The dummy msys dependency should have never been added instead the addons should have been fixed right from the beginning.

The mingw dependency is obsolete since #13140 and why should be keep something that doesn't work anymore?
If addons need mingw they have to provide is as their dependency not Kodi core.

I tried porting game.libretro.2048 to use msvc and it took me around 10 minutes. Situation before, didn't start, now it starts and we have debug symbols.

**Comment by Montellese** [comment]
> The dummy msys dependency should have never been added instead the addons should have been fixed right from the beginning.

This is easy to say but since the stance was (probably still is) not to touch / patch libretro cores it doesn't matter.

Back when I got game add-ons working on Windows I started by adding MinGW dependencies to every game add-on but that resulted in massive code duplication because every game add-on had to have the full build and setup instructions for MinGW. Therefore it was decided by the team that we would integrate this logic once into Kodi core so that every game add-on could simply reference it as a dependency and not have to care about how to download, compile, setup and install MinGW.

Concerning porting libretro cores to MSVC. This is great but only as long as upstream accepts these patches or we change our stance on not wanting to patch libretro cores manually.

TBH I don't really care how this is resolved because I don't have any time to contribute to it so I'm not gonna interfere. I mainly find it sad that people start ripping out stuff that was working and don't provide a fix for the things they break.

**Comment by Rechi** [comment]
How can an empty dependency be required by something?
If it is really needed by a libretro add the empty dependency to depends folder in the binary addon.

Currently we have https://github.com/kodi-game/game.libretro.2048/blob/master/depends/common/2048/CMakeLists.txt, change windows to use `add_library(${PROJECT_NAME} SHARED <files needed to be compiled>)` and adjust the install path. No single upstream change needed.

If anyone says I correct the mingw dependency (paths for the tools and name) I wait with merging otherwise I'll merge this PR as they are wrong in the current state.

**Comment by Montellese** [comment]
I don't remember the details for the msys dependency handling for binary add-ons since it's been a while. Furthermore back then there was no cmake buildsystem at all in Kodi so I don't know what has changed since the introduction of cmake in all of Kodi's buildsystem.

I also never tried to compile/fix game.libretro.2048. Back then I tried game.libretro.nestopia and some others and they did not compile with MSVC from Visual Studio 2012 because they used a lot of unix-specific headers and methods simply not available in MSVC and only available through MinGW. Maybe this has changed as well.

Concerning your argument that you will merge this because it is wrong in its current state I can only say that is has been right and worked perfectly fine until #13140 was merged. So your argument is like saying: oops someone introduced a bug into the video library in a previous PR so let's remove the whole video library because it is currently wrong. It's probably easier to revert part of #13140 now than after this has been merged as well.

**Comment by Rechi** [comment]
The cmake addon system was invented with the goal never use anything from Kodi core, as it may change. So #10964 should have never been accepted and instead every addon should have mingw as a dependency with all the logic needed for installing it.

Mingw in kodi core was at the beginning only need for building FFmpeg and libdvd, which was know by most windows developers, but not that binary addons are using it too. Because of that no one noticed it.

Regarding to your request to revert #13140, why should we keep something in Kodi core which isn't used any more for building core packages?

MSVC switching PRs are https://github.com/kodi-game/game.libretro.2048/pull/6 and https://github.com/kodi-game/game.libretro.mrboom/pull/4.

**Comment by garbear** [comment]
Kodi still contains the files:

* xbmc/project/BuildDependencies/DownloadMingwBuildEnv.bat
* xbmc/tools/buildsteps/win32/make-mingwlibs.bat
* xbmc/tools/buildsteps/windows/download-msys2.bat
* xbmc/tools/buildsteps/windows/make-mingwlibs.bat
* xbmc/tools/buildsteps/windows/make-mingwlibs.sh

and others. Should those be removed too?

**Comment by Paxxi** [comment]
So, I have had some time to think about the implications of this. I guess I was a driving force of having this in Kodi and not in the addons because of build speed and some disk usage.
I would love to get rid of it but not by forcing garbear to maintain a lot of build systems, then it would be better to fix this one the best we can.

**Comment by Voyager1** [comment]
AFAIK ffmpeg is still being built using mingw - did I miss anything? In which case these buildsteps are there to stay.

**Comment by Rechi** [comment]
Kodi only has msys no mingw anymore. Msys is still needed for FFmpeg, but only to run their configure script. MSVC is used for compiling the lib.

- xbmc/project/BuildDependencies/DownloadMingwBuildEnv.bat: only for Krypton jenkins compatibility
- xbmc/tools/buildsteps/win32/make-mingwlibs.bat: only for Krypton jenkins compatibility
- xbmc/tools/buildsteps/windows/download-msys2.bat: yes we only use msys2
- xbmc/tools/buildsteps/windows/make-mingwlibs.bat: should be named make-mysylibs.bat, but would need some jenkins config changes
- xbmc/tools/buildsteps/windows/make-mingwlibs.sh: should be named make-mysylibs.sh

File: cmake/addons/depends/windows/cmake/mingw/Toolchain_mingw32.cmake.in [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-set(CMAKE_SYSTEM_NAME Windows)
-
-set(CMAKE_FIND_ROOT_PATH @CMAKE_FIND_ROOT_PATH@ @CMAKE_INSTALL_PREFIX@ @MSYS_PATH@ @MINGW_PATH@)
-
-# specify the cross compiler
-set(CMAKE_C_COMPILER   @MINGW_PATH@/bin/gcc.bat)
-set(CMAKE_CXX_COMPILER @MINGW_PATH@/bin/g++.bat)
-set(CMAKE_AR           @MINGW_PATH@/bin/ar.bat CACHE FILEPATH "Archiver")
-set(CMAKE_LINKER       @MINGW_PATH@/bin/ld.bat CACHE FILEPATH "Linker")
-SET(CMAKE_RC_COMPILER  @MINGW_PATH@/bin/windres.bat)
```
Comment (Rechi) [inline cmake/addons/depends/windows/cmake/mingw/Toolchain_mingw32.cmake.in]: paths do not exist anymore, that's the reason I want to remove the files.

**Comment by afedchin** [comment]
> xbmc/tools/buildsteps/windows/make-mingwlibs.sh: should be named make-mysylibs.sh

should  be renamed to make-ffmpeg.(sh|bat) 
it builds ffmpeg only.

**Comment by Voyager1** [comment]
> should be renamed to make-ffmpeg.(sh|bat)

agreed. And thanks Rechi for the clarification on msys vs mingw.

**Comment by Montellese** [comment]
I never proposed to revert all of #13140. I suggest to revert the part that removes MinGW for easier maintainability of game add-ons.

Obviously adding custom `CMakeLists.txt` to the game add-ons for windows is an option BUT it has two major drawbacks over using MinGW:
* you have to completely rebuild the existing Makefile based buildsystem of the libretro core
* you have to keep the `CMakeLists.txt` in line with whatever changes are made to the libretro core when you update it

These two drawbacks don't apply to any other platform that we build these add-ons for just because we remove an existing and fully working functionality from Kodi core. If we were talking about some third party binary add-ons I'd probably be on your side but we are talking about binary add-ons written and maintained by @garbear who is part of Team Kodi.

**Comment by Rechi** [comment]
https://github.com/fetzerch/kodi-game-scripting/pull/22 moved installing the MinGW build system to the addons which need it.

**Comment by Rechi** [comment]
jenkins build this please

**Pull request merged**